### PR TITLE
Improve error handling for large JSON messages etc

### PR DESCRIPTION
**Pull request by tjamison**
Set maximum message size as optional compile-time variable
Ignore SIGPIPE in interactive example
When an unhandled error occurs on an SSL connection, attempt to
reconnect
Add RVI_ERR_JSON_PART as possible return value

Signed-off-by: Tatiana Jamison <tjamison@jaguarlandrover.com>

### CODE REVIEW

File: src/rvi.c [hunk 1]
```diff
@@ -640,11 +635,26 @@ int rviReadJsonConfig ( TRviHandle handle, const char * filename )
...
+    while (*buf != '\0' && *buf != delim && *buf++);
+    if (*buf != delim) {
+        buf = NULL;
+    }
+    return buf;
+}
+
 /* This utility function destructively reads from 'src' to populate '*json'
  * with a parsed json_t object. The remaining string (possibly 'src' exactly)
  * is copied into '*remainder'. The original 'src' must not be accessed again
```
Comment (gunnarx) [inline src/rvi.c]: the parameter **remainder** no longer exists but is mentioned in comment

File: src/rvi.c [hunk 2]
```diff
@@ -640,11 +635,26 @@ int rviReadJsonConfig ( TRviHandle handle, const char * filename )
...
  * by the calling function. */
-int rviReadJsonChunk( json_t **json, char *src, char **remainder )
+int rviReadJsonChunk( json_t **json, char *src, TRviRemote *remote )
```
Comment (gunnarx) [inline src/rvi.c]: Not sure if remainder and remote are the same thing?  But there's a fair amount of rewrite here - would be nicer to put it in a separate commit?

**Comment by gunnarx** [comment]
I took a look at the PR and made a few comments but I feel not able to understand the code fully enough to review the content and it also gets harder because of the many changes.  

All in all, too many unrelated changes at the same time (same commit even) for my taste, but you're the boss over this Tatiana, so it's really up to you as maintainer.

Since you are getting input from others that use the library, they too would probably benefit from smaller and atomic commits.  At this stage, merging soon and coming back to a stable state seems like the thing to do though, if you're not willing/able to split this up now.

But please consider  "git add -p" :-)

**Comment by tjamison** [comment]
Thanks for the feedback, @gunnarx. I'll keep doc refreshes in separate commits and keep future PRs smaller.
